### PR TITLE
[#48] [Backend] As a user I can delete a newsletter

### PR DESCRIPTION
**Pull request by tednguyendev**
#48

## What happened 👀
- Add `deleteNewsletter` to Newsletter Repository
- Add new route to delete a newsletter
- Add tests

## Insight 📝
N/A

### Before(Have 1 newsletter)

CleanShot 2023-06-14 at 14 28 01@2x

### Delete
CleanShot 2023-06-14 at 14 28 49@2x

### After(Have 0 newsletter)
CleanShot 2023-06-14 at 14 29 34@2x

### CODE REVIEW

File: nextjs-13/src/app/api/v1/newsletter/[id]/route.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  req: NextRequest,
+  { params }: { params: { id: string } }
+) {
+  return appHandler(req, async (currentUser) => {
+    try {
+      const result = await deleteNewsletter(params.id, currentUser.id);
+
+      if (result.count === 0) {
+        return NextResponse.json(
+          { message: 'Newsletter not exists' },
```
Comment (olivierobert) [inline nextjs-13/src/app/api/v1/newsletter/[id]/route.ts:17]: Minor english syntax issue: 

```suggestion
          { message: 'Newsletter does not exist' },
```

But we could be more generic:

```suggestion
          { message: 'Newsletter could not be deleted' },
```

**Review by olivierobert — CHANGES_REQUESTED** [review]
If you do not use a spellchecker yet, you should. It would allow you to catch syntax issues 🙈  Team members usually use Grammarly. There is even a VS Code extension for it.

File: nextjs-13/src/app/api/v1/newsletter/[id]/route.ts (hunk 1 above)
Comment (tednguyendev) [reply nextjs-13/src/app/api/v1/newsletter/[id]/route.ts:17]: Ahh my bad my bad 🙈 I do use Grammarly, but just for long writing text, not for the code much. I remember to check test names but forget checking response messages this time 🙈

Comment (tednguyendev) [reply nextjs-13/src/app/api/v1/newsletter/[id]/route.ts]: Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/113/commits)

**Review by phisith — APPROVED** [review]

**Pull request merged**